File: extract-fichas.py
import os
import re
import email
from email import policy
from email.parser import BytesParser
import json

def extract_email_body(filepath):
    """Extrair corpo do email"""
    with open(filepath, 'rb') as f:
        msg = BytesParser(policy=policy.default).parse(f)
    
    # Tentar pegar texto plain primeiro
    if msg.is_multipart():
        for part in msg.walk():
            content_type = part.get_content_type()
            if content_type == 'text/plain':
                body = part.get_content()
                return body
            elif content_type == 'text/html':
                body = part.get_content()
                # Limpar HTML básico
                body = re.sub(r'<[^>]+>', '', body)
                return body
    else:
        body = msg.get_content()
        return body
    return ""

def parse_ficha(body):
    """Extrair dados da ficha do corpo do email"""
    data = {}
    
    # Mapeamento de campos
    field_patterns = [
        ("name", r'Nome do cão\s*[:\-]?\s*([^\n]+)'),
        ("ownerName", r'Nome do tutor\s*[:\-]?\s*([^\n]+)'),
        ("breed", r'Raça\s*[:\-]?\s*([^\n]+)'),
        ("birthDate", r'Idade\s*[:\-]?\s*([^\n]+)'),
        ("sex", r'Sexo\s*[:\-]?\s*([^\n]+)'),
        ("castrated", r'Castrado\?\s*[:\-]?\s*([^\n]+)'),
        ("temperament", r'Nível de energia\s*[:\-]?\s*([^\n]+)'),
        ("size", r'Porte\s*[:\-]?\s*([^\n]+)'),
        ("doenca", r'Doença pré-existente\s*[:\-]?\s*([^\n]+)'),
        ("allergies", r'Alergias\s*[:\-]?\s*([^\n]+)'),
        ("feedingType", r'Tipo alimentação\s*[:\-]?\s*([^\n]+)'),
        ("feedingInstructions", r'Instruções\s*[:\-]?\s*([^\n]+(?:\n(?![A-Z]).*)*)'),
        ("feedingTimesPerDay", r'Vezes ao dia\s*[:\-]?\s*([^\n]+)'),
        ("feedingGramsPerMeal", r'Gramas por refeição\s*[:\-]?\s*([^\n]+)'),
        ("preferredActivities", r'Brincadeiras preferidas\s*[:\-]?\s*([^\n]+)'),
        ("vetName", r'Veterinário\s*[:\-]?\s*([^\n]+)'),
        ("allowPool", r'Piscina\?\s*[:\-]?\s*([^\n]+)'),
        ("allowPhotos", r'Fotos nas redes\?\s*[:\-]?\s*([^\n]+)'),
        ("serviceType", r'Serviços desejados\s*[:\-]?\s*([^\n]+)'),
        ("scheduledDays", r'Dias de frequência\s*[:\-]?\s*([^\n]+)'),
        ("ownerEmail", r'Email\s*[:\-]?\s*([^\n]+)'),
        ("ownerPhone", r'Telefone\s*[:\-]?\s*([^\n]+)'),
        ("ownerCpf", r'CPF\s*[:\-]?\s*([^\n]+)'),
    ]
    
    for field, pattern in field_patterns:
        match = re.search(pattern, body, re.IGNORECASE | re.MULTILINE)
        if match:
            value = match.group(1).strip()
            # Limpar
            value = re.sub(r'\s+', ' ', value)
            value = value.replace('\r', '').replace('\n', ' ')
            if value and value.lower() not in ['não informado', 'nenhuma informada', '']:
                data[field] = value
    
    return data

def main():
    anexos_dir = r'C:\Users\guede\Downloads\anexos'
    all_fichas = []
    
    for filename in os.listdir(anexos_dir):
        if filename.endswith('.eml'):
            filepath = os.path.join(anexos_dir, filename)
            print(f"Processando: {filename}")
            
            try:
                body = extract_email_body(filepath)
                data = parse_ficha(body)
                
                # Extrair nome do arquivo se não encontrou no corpo
                if 'name' not in data:
                    name_match = re.search(r'- (.+?) \(', filename)
                    if name_match:
                        data['name'] = name_match.group(1).strip()
                
                data['_source_file'] = filename
                all_fichas.append(data)
                print(f"  ✓ {data.get('name', '???')} - {len(data)} campos")
                
            except Exception as e:
                print(f"  ✗ Erro: {e}")
    
    # Salvar JSON
    output_file = r'C:\Users\guede\CascadeProjects\dog-hotel-app\all-fichas-extracted.json'
    with open(output_file, 'w', encoding='utf-8') as f:
        json.dump(all_fichas, f, ensure_ascii=False, indent=2)
    
    print(f"\n✅ {len(all_fichas)} fichas extraídas para: {output_file}")
    
    # Resumo
    print("\n📊 Resumo dos cães:")
    for ficha in all_fichas:
        name = ficha.get('name', '???')
        has_feeding = 'feedingType' in ficha or 'feedingInstructions' in ficha
        has_meds = 'doenca' in ficha
        has_allergies = 'allergies' in ficha
        print(f"  {name}: Alimentação={'✅' if has_feeding else '❌'} Meds={'✅' if has_meds else '❌'} Alergias={'✅' if has_allergies else '❌'}")

if __name__ == '__main__':
    main()

File: extract-fichas.js
const fs = require('fs')
const path = require('path')

// Simples extrator de campos de email
function extractFicha(filepath) {
  const content = fs.readFileSync(filepath, 'utf-8')
  
  const data = {}
  
  // Extrair campos com regex
  const fields = [
    ['name', /Nome do c[ãa]o\s*[:\-]?\s*([^\n]+)/i],
    ['ownerName', /Nome do tutor\s*[:\-]?\s*([^\n]+)/i],
    ['breed', /Ra[çc]a\s*[:\-]?\s*([^\n]+)/i],
    ['birthDate', /Idade\s*[:\-]?\s*([^\n]+)/i],
    ['sex', /Sexo\s*[:\-]?\s*([^\n]+)/i],
    ['castrated', /Castrado\?\s*[:\-]?\s*([^\n]+)/i],
    ['temperament', /N[íi]vel de energia\s*[:\-]?\s*([^\n]+)/i],
    ['size', /Porte\s*[:\-]?\s*([^\n]+)/i],
    ['doenca', /Doen[çc]a pr[ée]-existente\s*[:\-]?\s*([^\n]+)/i],
    ['allergies', /Alergias\s*[:\-]?\s*([^\n]+)/i],
    ['feedingType', /Tipo alimenta[çc][ãa]o\s*[:\-]?\s*([^\n]+)/i],
    ['feedingTimesPerDay', /Vezes ao dia\s*[:\-]?\s*([^\n]+)/i],
    ['feedingGramsPerMeal', /Gramas por refei[çc][ãa]o\s*[:\-]?\s*([^\n]+)/i],
    ['preferredActivities', /Brincadeiras preferidas\s*[:\-]?\s*([^\n]+)/i],
    ['vetName', /Veterin[áa]rio\s*[:\-]?\s*([^\n]+)/i],
    ['allowPool', /Piscina\?\s*[:\-]?\s*([^\n]+)/i],
    ['allowPhotos', /Fotos nas redes\?\s*[:\-]?\s*([^\n]+)/i],
    ['serviceType', /Servi[çc]os desejados\s*[:\-]?\s*([^\n]+)/i],
    ['scheduledDays', /Dias de frequ[êe]ncia\s*[:\-]?\s*([^\n]+)/i],
    ['ownerEmail', /Email\s*[:\-]?\s*([^\n]+)/i],
    ['ownerPhone', /Telefone\s*[:\-]?\s*([^\n]+)/i],
    ['ownerCpf', /CPF\s*[:\-]?\s*([^\n]+)/i],
  ]
  
  // Instruções é multi-linha, extrair especial
  const instrMatch = content.match(/Instru[çc][õo]es\s*[:\-]?\s*([\s\S]*?)(?=Vezes ao dia|Brincadeiras preferidas|Veterin[áa]rio|Piscina\?|$)/i)
  if (instrMatch) {
    let instr = instrMatch[1].trim()
    // Limpar
    instr = instr.replace(/\r\n/g, ' ').replace(/\n/g, ' ').replace(/\s+/g, ' ')
    if (instr && instr.length > 3 && !instr.toLowerCase().includes('não informado')) {
      data.feedingInstructions = instr.substring(0, 500) // Limitar
    }
  }
  
  for (const [field, regex] of fields) {
    const match = content.match(regex)
    if (match) {
      let value = match[1].trim()
      // Limpar
      value = value.replace(/\r/g, '').replace(/\n/g, ' ').replace(/\s+/g, ' ')
      if (value && value.length > 0 && 
          !value.toLowerCase().includes('não informado') && 
          !value.toLowerCase().includes('nenhuma informada') &&
          value !== '_') {
        data[field] = value
      }
    }
  }
  
  return data
}

const anexosDir = 'C:\\Users\\guede\\Downloads\\anexos'
const files = fs.readdirSync(anexosDir).filter(f => f.endsWith('.eml'))

const allFichas = []

console.log(`Processando ${files.length} fichas...\n`)

for (const filename of files) {
  const filepath = path.join(anexosDir, filename)
  
  try {
    const data = extractFicha(filepath)
    
    // Extrair nome do arquivo se não encontrou
    if (!data.name) {
      const nameMatch = filename.match(/- (.+?) \(/)
      if (nameMatch) {
        data.name = nameMatch[1].trim()
      }
    }
    
    data._source = filename
    allFichas.push(data)
    
    const hasFeeding = data.feedingType || data.feedingInstructions
    const hasMeds = data.doenca || data.allergies
    console.log(`✅ ${data.name || '???'}: Alim=${hasFeeding ? '✓' : '✗'} Meds=${hasMeds ? '✓' : '✗'}`)
    
  } catch (e) {
    console.log(`❌ Erro em ${filename}: ${e.message}`)
  }
}

// Salvar
const outputFile = 'C:\\Users\\guede\\CascadeProjects\\dog-hotel-app\\all-fichas.json'
fs.writeFileSync(outputFile, JSON.stringify(allFichas, null, 2))

console.log(`\n✅ ${allFichas.length} fichas salvas em ${outputFile}`)

// Resumo
console.log('\n📊 Resumo:')
for (const f of allFichas) {
  const fields = Object.keys(f).filter(k => !k.startsWith('_')).length
  console.log(`  ${f.name || '???'}: ${fields} campos`)
}
